openwisp_monitoring/device/base/models.py

```diff
@@ -253,6 +253,11 @@ def update_status(self, value):
         self.status = value
         self.full_clean()
         self.save()
+        # clear device management_ip when device is offline
+        if self.status == 'critical' and app_settings.AUTO_CLEAR_MANAGEMENT_IP:
+            self.device.management_ip = None
+            self.device.save()
```

Comment:
Try:

`self.device.save(update_fields=['management_ip'])`

It's more efficient.

Reply:
I was unaware of this. Thanks :smile: